### PR TITLE
Fixes reveal to AI  in headless clients and dedicated servers

### LINKED ISSUE #623: AI Detection Not Working On Dedicated Servers
**Arma 3 Version:** `1.88` (stable)
**CBA Version:** `3.90` (stable)
**ACRE2 Version:** `2.6.2` (stable)

**Mods:**

- CBA_A3
- acre2
- ACE3

**Description:**
If you have "acre_sys_core_revealToAI" enabled, ai can reveal players, if they are speaking in the "catchment area", depending on the sound-factor the player chooses (TAB). In local hosted missions (or for ZEUS on dedicated), where ai is local to the host, everything is working fine. but, if you are running the mission on a dedicated server, ai is not able to reveal players, because the eventhandler "acre_sys_core_onRevealUnit" is not registered on dedicated servers.

if you are looking into the code (acre2/addons/sys_core/XEH_postInit.sqf), you see at line 99 the eventhandler, wich is raised by the global event in "acre2/addons/sys_core/fnc_enableRevealAI.sqf" (line 64). BUT, in XEH_postInit (line 3) you'll see, that dedicated servers are "kicked out" of the script, therefore the eventhandler never gets registered on dedicated servers.

**Steps to reproduce:**
create a simple mission (one player unit and one enemy outside the line of sight) and run it on a dedicated server. teleporting the player behind a wall (or something), near to the enemy unit and speak in direct speech (like you are doing in normal missions to your team-mates). check, if the ai is turning to you and/or get his knowledge (enemy knowsAbout player) about the player. you will reconize, that the player is NOT revealed to the ai.

**Where did the issue occur?**
- Dedicated

**Placed Modules:**
- None

**RPT log file:**
- not needed?!

### PR DESCRIPTION
**Pull request by TheMagnetar**
**When merged this pull request will:**
- Fixes #623.
- This should affect HC as well.

### CODE REVIEW

File: addons/sys_core/fnc_enableRevealAI.sqf [hunk 1]
```diff
@@ -61,14 +61,14 @@ DFUNC(monitorAI_PFH) = {
...
                     _curUnit setVariable [QGVAR(lastRevealed), time, false];
-                    [QGVAR(onRevealUnit), [acre_player, _curUnit, ACRE_REVEAL_AMOUNT ] ] call CALLSTACK(CBA_fnc_globalEvent);
+                    [QGVAR(onRevealUnit), [acre_player, _curUnit, ACRE_REVEAL_AMOUNT], _curUnit] call CALLSTACK(CBA_fnc_targetEvent);
                 };
             };
         };
     } forEach _nearUnits;
 
-    if (GVAR(revealToAI) isEqualTo 0) then {
-        [(_this select 1)] call CBA_fnc_removePerFrameHandler;
```
Comment (Soldia1138) [inline addons/sys_core/fnc_enableRevealAI.sqf:71]: What is the reason for going to `==` instead of `isEqualTo`?

Comment (TheMagnetar) [reply addons/sys_core/fnc_enableRevealAI.sqf:71]: There was some discussion somewhere  (maybe ace slack?) that comparing numbers was still faster with ==. Jonpas reminded me of it in another PR

Comment (jonpas) [reply addons/sys_core/fnc_enableRevealAI.sqf:71]:
```
commy2 [19:25]
 Oh, and `!=` is faster than `!(isEqualTo)`.

commy2 [19:30]
 The only reason you should actively use IET is case sensitive comparission, the different interpretation of null, and comparing arrays.
Best with a comment explaining why for the first two of these.
Oh, or if the two data types could be different.
```

**Review by Soldia1138 — APPROVED** [review]
LGTM

**Review by jonpas — APPROVED** [review]

**Comment by jonpas** [comment]
Can't include in 2.6.3 due to incompatible bigger changes.

**Pull request merged**